### DISCUSSION TITLE
Help needed for error: CPU Time exceeded

### DISCUSSION THREAD

**Discussion by nerohoop** [supabase/supabase#27414] [Questions]
Hi team,

I'm trying to generate PDF with data from Supabase on Edge Function using pdfme. 
But I constantly ran into the following error

{
  "event_message": "CPU Time exceeded",
  "id": "b22d6fc3-20ac-407c-a7bc-f2b7cd3c58fa",
  "metadata": [
    {
      "boot_time": null,
      "cpu_time_used": 2100,
      "deployment_id": "sxlgbaogmfufqatjfsua_fa92285a-a09a-4562-9426-fc2569c5eded_41",
      "event_type": "Log",
      "execution_id": "f7c8b075-4887-4af5-b49b-f179e034aee8",
      "function_id": "fa92285a-a09a-4562-9426-fc2569c5eded",
      "level": "error",
      "memory_used": [
        {
          "external": 46876457,
          "heap": 24805688,
          "mem_check_captured": null,
          "total": 71682145
        }
      ],
      "project_ref": "sxlgbaogmfufqatjfsua",
      "reason": null,
      "region": "eu-west-2",
      "served_by": "supabase-edge-runtime-1.54.6 (compatible with Deno v1.43.0)",
      "timestamp": "2024-06-20T13:27:04.836Z",
      "version": "41"
    }
  ],
  "timestamp": 1718890029291609
}

I have upgraded my subscription plan but still ran into this error, is there any way to fix this issue?

Thanks

**Comment by GaryAustin1** [comment]
I don't believe CPU time changes based on subscription plan.
https://supabase.com/docs/guides/functions/limits#runtime-limits

You would have to make your task shorter.
